### DISCUSSION TITLE
How to disable all emails?

### DISCUSSION THREAD

**Discussion by mpvosseller** [pay-rails/pay#798] [Q&A] [answered]
Is there a straight forward way to disable all emails?

I previously followed the [configuration documentation and disabled the 4 types of emails described](https://github.com/pay-rails/pay/blob/master/docs/2_configuration.md#configuration) but it looks like [recently](https://github.com/pay-rails/pay/commit/36ae734f0718c852aa3856a18800c89488a982c2) 
`subscription_trial_will_end` and `subscription_trial_ended` emails were added.

Is there a way to disable all current and future emails?

**Comment by cjilbert504** [comment]
@mpvosseller - There's currently not way to disable all future emails. However, it is probably a good idea for this to be a feature. I'll make an issue for it and we should be able to get that added in easily enough. Thanks for bringing this up!

**Comment by mpvosseller** [comment]
@cjilbert504 Thanks. Also, after searching the code I think there are 3 email types currently not mentioned in the email configuration documentation:
- `payment_failed`
- `subscription_trial_will_end`
- `subscription_trial_ended` 

Did I miss any? 
As a first step I'd be happy to create a PR just add those to the documentation.

**Reply by cjilbert504** [reply]
That should be all but just in case it helps this is the current list:

```ruby
@@emails.payment_action_required = true
@@emails.payment_failed = true
@@emails.receipt = true
@@emails.refund = true
@@emails.subscription_renewing = ->(pay_subscription, price) {
(price&.type == "recurring") && (price.recurring&.interval == "year")
@@emails.subscription_trial_will_end = true
 @@emails.subscription_trial_ended = true
  ```

If you want to make a PR for the docs that'd be awesome! 

We can take care of adding the disabling of all emails functionality and make sure that we update the docs as well to reflect the ability to disable all emails.

**Comment by mpvosseller** [comment]
@cjilbert504 PR to update the documentation: https://github.com/pay-rails/pay/pull/800

**Reply by cjilbert504** [reply]
@mpvosseller - Thank you! Sorry for the delayed reply, I’ve been off the last couple of days while I move. I see it was merged, so thank you again for this one!

**Comment by cjilbert504** [comment]
@mpvosseller - Feature merged into `main`. See https://github.com/pay-rails/pay/pull/807#event-9271965859 will be part of next release. Thanks for bringing this up to us!

**Reply by mpvosseller** [reply]
Thanks @cjilbert504

**Comment by cjilbert504** [comment, marked as answer]
@mpvosseller - You bet, the new release is out now [6.5.0](https://github.com/pay-rails/pay/releases/tag/v6.5.0) :)
